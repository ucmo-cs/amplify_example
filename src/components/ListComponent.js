import React, { Component } from "react";
import { API, graphqlOperation } from 'aws-amplify';
import { listCars } from '../graphql/queries'
import { deleteCar } from '../graphql/mutations'

class ListComponent extends Component {

    constructor(props) {
        super(props)
        this.state = {
            cars: [],
            message: null
        }
    }

    componentDidMount() {
        this.loadCarList();
    }

    async loadCarList() {
        const result = await API.graphql(graphqlOperation(listCars));
        this.setState({cars: result.data.listCars.items })
    }

    async deleteCarEntry(carId) {
        const { cars } = this.state;
        const input = { id : carId };
        const result = await API.graphql(graphqlOperation(deleteCar, {input : input}));
        const deletedCarId = result.data.deleteCar.id;
        const updatedCars = cars.filter(car => car.id !== deletedCarId);
        this.setState({ cars: updatedCars });
    }

    editCarEntry(id) {
        window.localStorage.setItem("carId", id);
        this.props.history.push('/edit');
    }

    addCar() {
        window.localStorage.removeItem("carId");
        this.props.history.push('/add');
    }

    render() {
        return (
            <div>
                <h2 className="text-center">Car Details</h2>
                <button className="btn btn-danger" onClick={() => this.addCar()}> Add Car</button>
                <table className="table table-striped">
                    <thead>
                    <tr>
                        <th>Make</th>
                        <th>Model</th>
                        <th>Year</th>
                    </tr>
                    </thead>
                    <tbody>
                    {
                        this.state.cars.map(
                            car =>
                                <tr key={car.id}>
                                    <td>{car.make}</td>
                                    <td>{car.model}</td>
                                    <td>{car.year}</td>
                                    <td>
                                        <button className="btn btn-success" onClick={() => this.deleteCarEntry(car.id)}> Delete</button>
                                        <button className="btn btn-success" onClick={() => this.editCarEntry(car.id)}> Edit</button>
                                    </td>
                                </tr> )
                    }
                    </tbody>
                </table>
            </div>
        )
    }
}

export default ListComponent;
