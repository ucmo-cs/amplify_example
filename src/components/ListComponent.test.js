import { render, screen } from '@testing-library/react';
import ListComponent from './ListComponent'

jest.mock('aws-amplify');

describe('My test test suite', () => { 
    it('`true` should be `true`', () => {
        expect(true).toBe(true); 
    });
    it('`false` should be `false`', () => { 
        expect(false).toBe(false);
    }); 
    it('`1` should be `1`', () => { 
        expect(1).toBe(1);
    }); 
});

describe('List Component Test Suite', () => { 
    it('List Component Snapshot Test', () => {
        const wrapper = render(<ListComponent/>);
        expect(wrapper).toMatchSnapshot();
    });

});

