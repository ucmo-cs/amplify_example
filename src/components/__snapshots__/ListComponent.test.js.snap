// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component Test Shallow List Test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h2
          class="text-center"
        >
          Car Details
        </h2>
        <button
          class="btn btn-danger"
        >
           Add Car
        </button>
        <table
          class="table table-striped"
        >
          <thead>
            <tr>
              <th>
                Make
              </th>
              <th>
                Model
              </th>
              <th>
                Year
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h2
        class="text-center"
      >
        Car Details
      </h2>
      <button
        class="btn btn-danger"
      >
         Add Car
      </button>
      <table
        class="table table-striped"
      >
        <thead>
          <tr>
            <th>
              Make
            </th>
            <th>
              Model
            </th>
            <th>
              Year
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`List Component Test Suite List Component Snapshot Test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h2
          class="text-center"
        >
          Car Details
        </h2>
        <button
          class="btn btn-danger"
        >
           Add Car
        </button>
        <table
          class="table table-striped"
        >
          <thead>
            <tr>
              <th>
                Make
              </th>
              <th>
                Model
              </th>
              <th>
                Year
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h2
        class="text-center"
      >
        Car Details
      </h2>
      <button
        class="btn btn-danger"
      >
         Add Car
      </button>
      <table
        class="table table-striped"
      >
        <thead>
          <tr>
            <th>
              Make
            </th>
            <th>
              Model
            </th>
            <th>
              Year
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
